### ISSUE TITLE
iOS 15

### ISSUE THREAD

**Issue by extratone** [extratone/bilge#169]

**Comment by extratone** [comment]
# "[The iPad's inevitable Mac-like future is hiding in iPadOS 15](https://www.macworld.com/article/349404/apple-ipad-mac-floating-resizing-windows-keyboard-shortcuts-apps-shelf-quick-notes.html)"

**By *MORE COLOR
By Jason Snell JUN 24, 2021 4:15 AM PDT*** 
'Jun 24, 2021 at 06:15'
***
* * *

![5k iPad concept](https://www.macworld.com/wp-content/uploads/2021/06/5kipad-2.jpg?quality=50&strip=all&w=1024)

Jason Snell

In iPadOS 15, Apple is making another round of changes to iPad multitasking. It would be easy to consider this yet another spin on the merry-go-round as Apple struggles to figure out how to break the iPad out of the iPhone’s fundamental one-app-at-a-time interface.

But I don’t. Instead, I think Apple is finally assembling all the pieces of the puzzle that will allow the iPad to become more Mac-like than ever before—all the while retaining its unique position in Apple’s ecosystem.

Is Apple waiting for the right moment to unveil proper external-display support on the iPad, perhaps alongside the release of a new Apple display? I’m not saying it will definitely happen, but [I want to believe](https://go.redirectingat.com/?id=111346X1569486&url=https://knowyourmeme.com/memes/i-want-to-believe&xcust=1-3-349404-0-0-0|xid:fr1624878695674ehi&sref=https://www.macworld.com/article/349404/apple-ipad-mac-floating-resizing-windows-keyboard-shortcuts-apps-shelf-quick-notes.html). So let’s look at the evidence.

## A window into the future

Apple has been flirting with the idea of putting floating windows on the iPad for a few years now. For example, Slide Over is a floating window attached to one side of the screen. Picture in Picture floats above apps but must be placed in a corner. Two years ago, Apple introduced the concept of multiple app windows—but they were really just multiple instances of an app, running in existing app frames. (A Microsoft Word file in full screen and a different Word file in a different Split View, for example.)

But with iPadOS 15, things are getting more interesting on the windowing front. Yes, the iPad’s new [Globe-key shortcuts](https://go.redirectingat.com/?id=111346X1569486&url=https://sixcolors.com/post/2021/06/think-globally-the-ipads-new-universal-keyboard-shortcuts/&xcust=1-3-349404-0-0-0|xid:fr1624878695674cgj&sref=https://www.macworld.com/article/349404/apple-ipad-mac-floating-resizing-windows-keyboard-shortcuts-apps-shelf-quick-notes.html) seem to point to more sophisticated multitasking to come, but what really caught my eye is a more old-fashioned Command-key shortcut that Apple has imported from the Mac: next App Window, executed by holding down Command and the tick-mark (`) key.

![iPadOS 15 multitasking](https://b2c-contenthub.com/wp-content/uploads/2021/06/ipados15-multitasking.jpg?quality=50&strip=all&w=1200)

Apple

This is the Mac’s “cycle through windows” keyboard shortcut that became a part of my muscle memory in the early days of Mac OS X, and I still use it all the time. On iPadOS 15, pressing Globe-tick will move focus among different visible apps when in Split View and Slide Over, so you can choose which app will receive keyboard input. Pressing Command-tick cycles through multiple windows within a single app.

Add to that the introduction of the iPad’s newest interface element, the Shelf, a visual representation of an app’s currently open windows. This feels like Apple adding complexity in order to add clarity, and I’m not entirely sold on it conceptually. Regardless, the Shelf’s very existence shows that Apple is trying to find ways to make multi-window interfaces in iPad apps discoverable and usable.

And then there are iPadOS 15’s new additions to the growing family of iPad windows. First off, there’s the new “floating center window,” which appears (among other places) when you create a new message in Mail. It’s a window that initially appears hovering above the app that spawned it, but it’s got Apple’s new three-dot windowing interface at the top that will let you drag it around (hmm, I wonder why would you need a drag area for iPad windows?), or move it to its own full screen, Split View, or Slide Over location.

![iPadOS 15 quick note](https://b2c-contenthub.com/wp-content/uploads/2021/06/ipados15-quicknote.jpg?quality=50&strip=all&w=1200)

Apple

What separates the floating center window from just being a window from a Mac app? Only iPadOS acting as a policeman, limiting where you can place it. Otherwise, it’s a floating window of the kind Mac users have known for decades.

Which brings us to Quick Note, a new feature coming to iPad, iPhone, and Mac this fall that is implemented in iPadOS as… a floating window. Yes, it behaves like a Picture in Picture window–you can only place it in the corners, or you can park it off the side of the screen with a translucent arrow widget indicating that it’s ready to be resurfaced.

Even if you discount all the other evidence that Apple is building a foundation for free windowing on the iPad, it’s _extremely_ hard to look at Quick Note and believe anything else. It’s an independent floating window layered above other apps. Windowing on the iPad isn’t just inevitable. It’s already here.

## Apps of all shapes and sizes

One of the biggest arguments against offering Mac-style windowing on the iPad is that iPad (and iPhone) apps aren’t built to be infinitely resizeable like Mac windows are. 

There’s truth in that, though it’s not true as it once was. Modern iOS apps have to adapt to numerous screen sizes and orientations. And once again, the Mac shows the way: If you use an iOS app on an M1 Mac, you’ll see that they’re resizable—with limits.

I doubt that any iPad windowing scheme would offer pixel-perfect resizing of the sort the Mac offers. But imagine resizing a window and having it snap to a bunch of different aspect ratios and sizes—corresponding to the dimensions of various supported Apple devices.

iOS apps can be chameleons. I think I’d actually prefer to run the iPhone layout of some apps on my iPad. Some apps are a bit ridiculous on a 12.9-inch display, but quite delightful when displayed in the compact frame of an iPhone 12 mini.

## It’s got menus, bar optional

It’s not just windows. Apple is also rebuilding the Mac menu bar on the iPad, and it’s doing it in plain sight. Hold down the Globe or Command keys in iPadOS 15, and you’ll see an overlay that lists all the available keyboard shortcuts—organized in a quite familiar pattern that begins with _File_ and _Edit_. You don’t even need to use keyboard shortcuts–tapping on any command will execute it.

![iPad OS 15 quick notes](https://b2c-contenthub.com/wp-content/uploads/2021/06/iPadOS-15-quick-notes.jpg?quality=50&strip=all&w=1200)

Apple

How are there hidden menus inside iPad apps? The answer lies with Catalyst, which enables iPad app developers to bring their apps over to the Mac. Now iPadOS 15 makes that extra work visible on the iPad, too. (Catalyst apps can also open multiple windows, and on the Mac they appear as regular, floating independent windows. Hmm.)

Would Apple ever offer a proper Mac-style menu bar on the iPad? I think it might, in the proper contexts. Perhaps only when a keyboard and pointing device are attached, or only on an external display that has to be entirely driven by those input methods. But (as my fever dream of a 5K iPad mock-up shows), it’s not as if the iPad isn’t already most of the way there. The iPad’s status bar displays time, date, and a bunch of icons already–they’d be at home in a Mac-style menu bar. Toss in icons for Control Center and Notification Center if you want.

## Not now, but soon

All the pieces are there: a proper mouse-driven pointer, multiple windows per app, menu bar items, and apps that can adjust to different sizes. It’s not hard to imagine, using almost entirely existing iPad interface elements, what a bunch of different iPad apps floating on a 5K display might look like. That’s what I did—and you can see it at the top of this article.

Would I like to plug my M1 iPad Pro into a 5K external display and see an interface like the one illustrated above? Why, yes. Yes I would. As Apple has repeatedly made clear, what sets the iPad apart from other devices is that it’s supremely adaptable. It can be a naked touch tablet, but you can add an Apple Pencil, a keyboard, or a Magic Keyboard with a trackpad.

Adding a big, bright display and filling it with app windows doesn’t break the premise of the iPad. It fulfills it. I hope Apple sees it that way, too.

_Note: When you purchase something after clicking links in our articles, we may earn a small commission. Read our_ _[affiliate link policy_](/about/affiliate-link-policy) _for more details._

Jason was lead editor at Macworld for more than a decade and now writes about Apple at [Six Colors](https://sixcolors.com) and podcasts at [Relay FM](http://www.relay.fm) and [The Incomparable](https://www.theincomparable.com). 


-"[The iPad's inevitable Mac-like future is hiding in iPadOS 15](https://www.macworld.com/article/349404/apple-ipad-mac-floating-resizing-windows-keyboard-shortcuts-apps-shelf-quick-notes.html)"

**Comment by extratone** [comment]
# "[iOS 15 and iPadOS 15 preview](https://www.theverge.com/2021/6/30/22556236/ios-15-ipados-iphone-ipad-software-update-public-beta-preview-apple)"

**By *Chaim Gartenberg*** 
'Jun 30, 2021 at 12:30'
***
![](https://cdn.vox-cdn.com/thumbor/beX3BII3WZHxXG0DO6sv4kJXSDk=/0x0:2040x1360/1200x675/filters:focal\(843x746:1169x1072\)/cdn.vox-cdn.com/uploads/chorus_image/image/69525168/cgartenberg_210629_4653_001.0.jpg)

iOS 15 and iPad 15 are [kicking off their public betas today](https://www.theverge.com/22557308/ios-ipados-15-public-beta-apple-install), and after a few weeks with the developer betas of the new software, Apple’s OS updates feel like more of a grab bag of new features than ever before. 

A major rethinking of either platform, this year’s updates are not. The two updates were clearly born in 2020’s norm-shattering pandemic. The feature list at WWDC and on Apple’s website wears last year’s remote-first influences firmly, from the heavy emphasis on FaceTime features to a better system for corralling notifications into “work” and “personal” buckets. 

Of course, Apple’s release cadence and insistence on major updates coming just once a year means that some of these features are arriving in what will — hopefully — be a post-pandemic return to life this fall. So it’ll be interesting to see how things like SharePlay, one of the update’s marquee features, actually shake out once people have the option to watch movies and listen to music together in person again.

Other parts of iOS and iPadOS 15, like the overhauled Safari app or the new Maps app, feel like the kinds of more noticeable changes generally associated with Apple’s major software updates, but they’re sporadic. 

The result is a software update that feels a lot quieter than Apple’s usual releases, one that — at least for now — looks to improve smaller things behind the scenes than rebuild things from the ground up.

![](https://cdn.vox-cdn.com/thumbor/M_1Y1hkKEwBLCuN0GZM4jPb9FeY=/0x0:2040x1360/1200x0/filters:focal\(0x0:2040x1360\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22689245/cgartenberg_210629_4653_002.jpg) The iOS 15 homescreen is largely unchanged from iOS 14. 

## **FaceTime and SharePlay**

The flashiest feature coming to the fall updates is SharePlay, a new Apple-wide system built on top of FaceTime for sharing TV shows, movies, music, and podcasts with friends and family even when you’re not in the same room. It’s also the most pandemic-inspired feature, an Apple product-based spin on the [countless watch party apps and services](https://www.theverge.com/21307583/watch-videos-movies-online-friends-streaming-netflix-hulu-amazon-scener-extensions) that sprung up to replace movie nights over the past year. 

SharePlay shows off how well Apple’s hardware and software services all work together: watching a TV show or listening to an album over FaceTime is seamless. But it also highlights the height of Apple’s walled garden: developers have to choose to use SharePlay — which is missing big names like Netflix and YouTube right now — and it only works on Apple hardware, despite the expansion of FaceTime and the Apple TV app to other platforms. 

![](https://cdn.vox-cdn.com/thumbor/cWkrmK5-4Xr7UpjowbLfHYs_ehQ=/0x0:2686x1314/1200x0/filters:focal\(0x0:2686x1314\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22691037/Screen_Shot_2021_06_30_at_10.11.02_AM.png) _SharePlay in action._ Image: Apple

And since content is streamed locally, everyone watching or listening has to have access to it, meaning that you can’t share an episode of _Ted Lasso _with a friend who’s not a TV Plus subscriber, nor can you both watch the same movie if only one of you has purchased it from iTunes. 

FaceTime is also getting a wide variety of long-overdue updates that help cement it as a more viable video chatting application: a sorely needed grid view, screen sharing, a portrait mode to blur your messy background, and the ability to FaceTime with Windows and Android users (in web browsers) thanks to sharable links. 

These are all badly needed features for FaceTime, but they’re also the sort of thing that remains bewildering to have had to wait until September 2021 for, especially given the reliance on video calling over the past year and a half.

Unlike SharePlay, though, the FaceTime improvements feel like the kind of thing that will remain front and center even as things start to go back to normal, although the Apple product focus and more limited feature set when compared to professional solutions like Zoom or Microsoft Teams mean that FaceTime won’t be making a play for business meetings any time soon. 

## **Focus up**

As is traditional for an iOS update, Apple has done some tinkering with notifications on iOS 15. Some of the smaller notifications are bigger now, and feature contact images — which might be the push to get people to actually add photos to their contacts, something that doesn’t generally exist in the real world outside of an Apple presentation in my experience. And apps that you care less about can be filtered to a new summary mode that’s delivered several times a day, instead of pinging you for each notification right away. 

But the bigger addition is the new Focus feature, and after only a few weeks of using it, it’s already one of my favorite features on iOS in years. On the surface, Focus is an expansion of Apple’s existing Do Not Disturb feature, but instead of a blanket mute, Focus allows you to select specific apps and contacts to share notifications at specific times or specific triggers. 

![](https://cdn.vox-cdn.com/thumbor/_XEOGXJjrBJ53jBTcE4jV-rLQ9A=/0x0:3360x2100/1200x0/filters:focal\(0x0:3360x2100\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22691389/focus.png) _Focus settings (left), the expanded Do Not Disturb menu (middle), and Focus-siloed notifications (right)._

A “work” Focus, for example, can be set to activate when you’re on the clock and mute all notifications except from your email, Slack, and calendar apps, then automatically switch off when you’re off work. Focus modes can be triggered by specific times, locations (like when you get to your office or home from work) or when opening a specific app. 

Focus gives you real control over filtering notifications

So far, I’ve mostly just been using a “personal” focus for weekends and evenings to automatically mute any work Slacks and emails until the morning, which has been absolutely delightful. Apple’s machine learning also tries to learn from how you use your phone — for example, it suggested that I add a sports app I use a lot during “personal” time to the whitelist. 

And while Focus can be used in conjunction with Apple’s Screen Time feature, it’s not a totally cohesive system: for example, there’s no way to automatically disable work apps like Slack when in a “personal” Focus mode. It just shuts off notifications. 

Focus also lets you assign specific homescreen pages to each Focus mode — including widgets — for even more customized experiences. It’s not a perfect system, since iOS won’t let you have duplicate icons for apps (so you can’t have Apple Music visible on both a “personal” and a “work” homescreen, for example). But I’m a big fan of the feature, since it gives me a good reason to actually set up homescreens with widgets, and helps curb the instinct to tap the Slack icon when I’m supposed to be offline. 

There’s also a new kind of notification to go with Focus and the morning summary feature, called “Time Sensitive Notifications,” which can override specific notification filtering features, so you don’t miss urgent alerts from your bank, for instance. Apple has specific rules for when developers can use these, but we’ll have to wait to see how they’re actually implemented when apps get updated this fall. 

## **Safari explores a new look**

![](https://cdn.vox-cdn.com/thumbor/eOUN8sed_GJYWGBxhVNjcbwIaC0=/0x0:3360x2100/1200x0/filters:focal\(0x0:3360x2100\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22691391/safari.png) Safari’s new design, with the rearranged URL bar.

Safari has been entirely redesigned in iOS 15. In terms of material changes to iOS, this is the biggest, and it’s coming to what might be the most essential application on the platform. 

For the iPhone in particular, that means an emphasis on a one-handed design that moves the URL bar to the bottom of the screen, adds a swiping gesture interface, and a tab grouping feature (which is also coming to iPadOS and macOS Monterey). 

I’m still getting used to it, even with a few days of use under my belt, and of all the additions in the latest software update, I suspect it’ll be the most polarizing. Over a decade of muscle memory has trained my brain to reach up for the menu bar on smartphones. I get Apple’s motivation in moving it to the bottom, making it easier to reach on the increasingly large phones it makes and putting the actual content of devices front and center at the top of the screen, it’s still a change that’ll require an adjustment period. 

![](https://cdn.vox-cdn.com/thumbor/lBPqBrPVYuf7-dI9-M8fE5mLx1U=/0x0:2040x1360/1200x0/filters:focal\(0x0:2040x1360\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22689246/cgartenberg_210629_4653_003.jpg) iPadOS 15 allows for widgets on the homescreen now.

## **iPad multitasking: what’s a computer?**

Overall, iPadOS 15 gets a lot of the same improvements as the iPhone update. There’s a few iPad-specific features, though, starting with two major features from iOS 14 that were oddly missing last year: widgets on the homescreen and the App Library (which comes with a spiffy animation when you open it). I still can’t explain why Apple didn’t add those features last year, but they’re extremely welcome now. 

Widgets, as expected, work even better on the iPad than they do on the iPhone, given the bigger homescreen. It’s not as big of a sacrifice to give up a quarter of your homescreen for widgets when there’s still so much extra space on the screen. Apple has also added even _bigger_ widget sizes, the biggest size of which is roughly the same area as an entire iPhone 12 display when viewed on a 12.9-inch iPad Pro. Despite the massive size, Widgets still have the same functionality as they do on the iPhone, meaning that they’re focused on glanceable information rather than more interactive mini applications. 

Apple is also continuing to refine multitasking on the iPad with two new additions: a multitasking icon (three dots at the top of the display) that makes it easier to use the different split-view and slide-over modes it introduced in 2019, and a “shelf” that shows all the open windows for a particular app when you open it or tap the multitasking icon. 

![](https://cdn.vox-cdn.com/thumbor/SAw8SB-VHOhzoIQM3jMG1BDaVeE=/0x0:2732x2048/1200x0/filters:focal\(0x0:2732x2048\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22691086/IMG_0008.PNG) _Multitasking on iPadOS 15 is still... a lot._

The new multitasking setup is better, albeit still slightly confusing. Putting a single app into split view using the new menu punts you back to the homescreen to select a second app (or the same app again) to view in split-screen mode. Apps can be dragged and dropped when you’re in the app switching window to create new split-screen or full-screen combinations. The multitasking dots light up to show which app in split-screen is in focus, which is incrementally better than the bar from before. And swiping down on the multitasking dots is now an almost home-button-like experience, closing that app and letting you select a new one. 

There are still a lot of ways to move things around and arrange things and keyboard shortcuts available — so much so that Apple has a new menu when you hold the globe key to view system and multitasking shortcuts.

But there’s still plenty of friction here. iPadOS remains unpredictable: I’m never 100 percent sure what version of an open app or app windows will open when I tap in on my homescreen. Just opening a new window is still an opaque process that involves dragging and dropping things around in split view. Slide-over panels still live in their own, separate confusing world. Split view is still frustratingly rigid, letting you have exactly two apps open with a third visible as a slide-over panel, instead of any other configuration (like one large app on the left and two smaller ones on the right). 

An iPad operating system with iPad apps — and that’s just how Apple likes it

Ultimately, the new multitasking and split-screen views are a refinement of the older system, rather than some grand new paradigm for how to use an iPad. Those who like the iPad’s software abilities will likely find the new additions and enhancements to those modes nice. But those who were hoping that iPadOS 15 would offer a wildly overhauled windowing system — especially in the wake of Apple’s M1 upgrade in the latest iPad Pro — are going to be disappointed. Apple may do some additional tweaks or changes before iPadOS 15 launches this fall, but it’s unlikely to do any sweeping design changes for this update. iPadOS is still firmly an iPad operating system with iPad apps, and that seems to be how Apple likes things. 

Also new is a Samsung-like “Quick Notes” feature that lets users swipe up from the bottom corner of the display or hit a keyboard shortcut and scribble down a thought, highlight text on a website, or add a link for context. Quick Notes get saved to their own category on the Notes app, and can be accessed from other Apple devices, too. They’re useful, although like many iPadOS 15 features, it’s not something you’d likely figure out how to use without outside prompting. 

## **The best of the rest**

As is always the case, there are dozens of bigger and smaller features coming in iOS and iPadOS, too. Here are some of the more notable ones to check out on the public beta:

Live Text might be the most technically impressive part of the updates, letting you point your camera at any handwritten or typed text to grab a phone number or email address, translate it, or pull it directly into a text field. Apple has quietly made this a system-level feature — meaning that it’s available whenever you’re viewing an image, be it in the live camera feed, an image on the web, or directly from the keyboard. 

Live Text might be the most technically impressive update ![](https://cdn.vox-cdn.com/thumbor/B8ClwJEIj9Pw2f509_hAPvg0uHI=/0x0:1170x1294/1200x0/filters:focal\(0x0:1170x1294\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22691400/text_image.jpg) Live Text on iOS 15 can capture text directly from a photo feed.

An interesting new addition on both operating systems is a new “Shared with You” feature, which shows photos, Apple Music songs, links, Apple TV Plus content, and podcasts that were shared over iMessage in dedicated rows within those apps. The theory is that if someone texts you a nice photo or their favorite playlist, you’ll be able to see it (and respond to it) in the Photos or Music apps. The feature is unsurprisingly entirely limited to Apple services for now, so sending a Spotify song to someone won’t appear in Spotify. It also might be overestimating how often users send links to TV Plus shows to each other, but showing shared links and photos are admittedly useful (at least, when the feature works properly).

The Weather app has an overhauled design that adds a vertical, Dark Sky-style 10-day forecast and Dark Sky-style maps for temperature and precipitation. It looks nice. Notes and Reminders are both getting hashtags for categorizing and organizing, and the Memories feature in Photos is getting completely overhauled. 

Lastly, you can now put your Memoji avatars in outfits. I have no opinion on this.

![](https://cdn.vox-cdn.com/thumbor/E0zJ2M2dfqUhKWBfYTM6rGIRf9E=/0x0:1200x800/1200x0/filters:focal\(0x0:1200x800\):no_upscale\(\)/cdn.vox-cdn.com/uploads/chorus_asset/file/22691105/Untitled.png) ¯\\_(ツ)_/¯ 

A lot of the other flashy new features are things we can’t test yet. Apple’s beta is only rolling out new maps to the San Francisco Bay Area to start (although it’ll be in far more locations when the final software is out this fall). IDs and keys in Wallet will also have to wait for broader support from state governments, hotels, and workplaces before we can see how those work. 

There are also a lot of features that will have to wait for third-party developers, like Focus statuses or SharePlay in third-party apps, which also have to wait for app updates. 

* * *

The most notable thing from my time with the betas, though, is how easy it is to miss the newer features. Outside of the most in-your-face updates like the larger notifications or the new Safari design, most of the updates here are quieter, behind-the-scenes features.

They’re things that give you more options and ways to use your devices, but they’re just that — options. Focus modes are great for those who want to use them and integrate them into their workflow, but iOS doesn’t go out of its way to advertise them or push you to use them. It’s a mature way of looking at software, and one that’s nice to see, even if it means that iOS 15 and iPadOS 15 aren’t the most exciting updates ever. 

_Photography by Chaim Gartenberg / The Verge_


-"[iOS 15 and iPadOS 15 preview](https://www.theverge.com/2021/6/30/22556236/ios-15-ipados-iphone-ipad-software-update-public-beta-preview-apple)"

**Comment by extratone** [comment]
# Raw iCloud Sharing Links for (some Siri Shortcuts)

'''
https://www.icloud.com/shortcuts/8cad4040980e438cb8888a196669cb73
https://www.icloud.com/shortcuts/1685c35c9cea40768e48fc07c4f35771
https://www.icloud.com/shortcuts/b86715a9e3414940aef6c1e248b5c7a9
https://www.icloud.com/shortcuts/037b4dd621bc4eeeafd1e2b99d1d1e8d
https://www.icloud.com/shortcuts/5f887285b6bc45cd9d3182442a6c1570
https://www.icloud.com/shortcuts/971289dfa1b84682878c4e3b8104ff14
https://www.icloud.com/shortcuts/41418cff3f0f46c2ae4cafff4f393a4e
https://www.icloud.com/shortcuts/8ecd49dc88d447d790ef66c9dff48b95
https://www.icloud.com/shortcuts/a0c7ea117f654360b90eaebc74fae282
https://www.icloud.com/shortcuts/5a6ef961edee424e95ca887026d976e5
'''
